library/std/src/collections/hash/map.rs

```diff
@@ -1827,6 +1895,66 @@ where
     }
 }
 
+#[unstable(feature = "map_into_keys_values", issue = "55214")]
+impl<K, V> Iterator for IntoKeys<K, V> {
+    type Item = K;
+
+    #[inline]
+    fn next(&mut self) -> Option<K> {
+        self.inner.next().map(|(k, _)| k)
+    }
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.inner.size_hint()
+    }
+}
+#[unstable(feature = "map_into_keys_values", issue = "55214")]
+impl<K, V> ExactSizeIterator for IntoKeys<K, V> {
+    #[inline]
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+}
+#[unstable(feature = "map_into_keys_values", issue = "55214")]
+impl<K, V> FusedIterator for IntoKeys<K, V> {}
+
+#[unstable(feature = "map_into_keys_values", issue = "55214")]
+impl<K: Debug, V: Debug> fmt::Debug for IntoKeys<K, V> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_list().entries(self.inner.iter()).finish()
```

Comment:
It would be more appropriate to print just the keys. I think as implemented this would print both keys and values, but the values are inaccessible via this iterator.

Reply:
Yeah, you're right. I fixed this for HashMap. BTreeMap ones have `#[derive(Debug)]` on them (because that was also like that for their `Keys` and `Values` structs), and I hope it will print it correctly with the map in the iterator implementation, but not so sure.

I tried to implement `Debug` manually for them as well, but `IntoIter` struct of BTreeMap doesn't have `iter` function [like HashMap](https://github.com/rust-lang/rust/blob/8b26609481c956a666f9189738f1ba611078e1ab/library/std/src/collections/hash/map.rs#L1065-L1071) and it wasn't possible to implement it(although maybe it is, but I just don't know how). Do you think this is okay like this, or should we implement `iter` for IntoIter first and then manually implement them as well?

Reply:
It seems like there is a lot of inconsistency in this area. Would you be interested to follow up with a separate PR to resolve the inconsistency between HashMap and BTreeMap with regard to iterator Debug impls?

Reply:
Sure, I can work on that as a follow-up. Created #75297 to track this.